docs/architecture.md

```diff
@@ -1,14 +1,102 @@
-### cozy-client architecture
+### Cozy-client Architecture
 
 This is an overview of the cozy-client architecture:
 
 ![Architecture schema](./architecture.png)
 
-The architecture schema can also be accessed [online here](https://whimsical.co/6AJnnS7v5ePRcuKZysPAF5).
-It's created on https://whimsical.co/ — if you need to edit it, ask for an invite from one of the projects maintainers.
+<!-- The architecture schema can also be accessed [online here](https://whimsical.co/6AJnnS7v5ePRcuKZysPAF5).
+It's created on https://whimsical.co/ — if you need to edit it, ask for an invite from one of the projects maintainers. -->
 
-### Missing parts
+To explain the architecture, we will see how a request for data flows
+through cozy-client and how finally the UI is updated.
 
-- Init
-- Associations
-- Mutations
+### Query defitions
+
+To query data, we create a *query definition*: an object describing documents to fetch.
+
+```
+{
+  doctype: "io.cozy.todos",
+  selector: {
+    "finished": true
+  }
+}
+```
+
+It is typically created via the helper `Q` that provides a fluid interface
+to create a *Query Definition*.
+
+```jsx
+const qdef = Q('io.cozy.todos').where({ finished: true })
+```
+
+Then we need to execute this query, fetch the data, and storing it.
```

Comment:
```suggestion
Then we need to execute this query definition in order to fetch the data.
```

What do you want to say when you say : "storing it"? You mean in the redux store?

Reply:
Yep